### PR TITLE
Stack Definition Reference

### PR DESCRIPTION
**Pull request by tjholm**

### CODE REVIEW

**Review by davemooreuws — APPROVED** [review]

File: docs/reference/project/stack.mdx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+## Full Example
+
+```yaml
+name: example-image-processor
+
+functions:
+	get-processed-image:
+		handler: functions/get.ts
+  store-image:
```
Comment (medgar-nitric) [inline docs/reference/project/stack.mdx:25]: Viewing this file in Github show red blocks:
https://github.com/nitrictech/docs/blob/9525e6f7045bc0c5b8f7c1d5bf87e88ad5676edd/docs/reference/project/stack.mdx
Not sure if this is intentional or a display issue on Github.

File: docs/reference/project/stack.mdx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+sites:
+  app:
+    path: app
+    assetPath: build
+    buildScripts:
+      - yarn build
+
+apis:
+  api: api.yaml
```
Comment (HomelessDinosaur) [inline docs/reference/project/stack.mdx:61]: Could this be clearer that "api" is the name of the api by renaming it to something like "image-api"?

File: docs/reference/project/stack.mdx (hunk 1 above)
Comment (tjholm) [reply docs/reference/project/stack.mdx:25]: Replaces tabs with spaces

File: docs/reference/project/stack.mdx (hunk 2 above)
Comment (tjholm) [reply docs/reference/project/stack.mdx]: done

**Review by HomelessDinosaur — DISMISSED** [review]

File: docs/reference/project/stack.mdx [hunk 3]
```diff
@@ -0,0 +1,72 @@
+A nitric stack is described by its definition in a `nitric.yaml` file. A stack represents a collection of resources that operate with each other and are deployed together.
+
+## Properties
+
+| **name**    | **description**                                              | **type**                           |
+| ----------- | ------------------------------------------------------------ | ---------------------------------- |
+| name        | A descriptive name of the stack                              | `string`                           |
+| functions   | zero or more functions to be deployed                        | map of [function](./functions)     |
+| containers  | zero or more containers to be deployed                       | map of [container](./containers)   |
+| topics      | zero of more topics to be deployed                           | map of [topic](./topics)           |
```
Comment (jyecusch) [inline docs/reference/project/stack.mdx:10]: I think queues are missing from this list and the placeholder docs

**Review by jyecusch — CHANGES_REQUESTED** [review]

File: docs/reference/project/stack.mdx (hunk 3 above)
Comment (tjholm) [reply docs/reference/project/stack.mdx]: done

File: docs/reference/project/stack.mdx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+      topics:
+        - process-trigger
+
+topics:
+  process-trigger: {}
+
+buckets:
+  raw-images: {}
+  processed-images: {}
+
```
[the remark this answers is not in the document]
Comment (tjholm) [reply docs/reference/project/stack.mdx:43]: have put this in with placeholder reference document as well

**Review by jyecusch — APPROVED** [review]

**Review by medgar-nitric — APPROVED** [review]

**Pull request merged**